guppylang/decorator.py

```diff
@@ -61,6 +61,12 @@ class _Guppy:
     def __init__(self) -> None:
         self._modules = {}
 
+    @overload
+    def __call__(self, arg: PyFunc) -> RawFunctionDef: ...
```

Comment:
These got rid of a load of errors on `@guppy(builtins)` but we could say that typechecking `builtins.py` or any other guppy file is such a lost cause that we shouldn't bother....?

Reply:
Yeah, this is a difficult topic. I think for us it's acceptable to disable mypy for `builtins.py`, but actual users will run into the same problem when they run mypy on their code. Also see #277

Happy to keep these `__call__` overloads if they help 👍